sopel/bot.py

```diff
@@ -305,18 +305,26 @@ def say(self, text, recipient, max_messages=1):
 
             recipient_id = Identifier(recipient)
 
-            if recipient_id not in self.stack:
-                self.stack[recipient_id] = []
-            elif self.stack[recipient_id]:
-                elapsed = time.time() - self.stack[recipient_id][-1][0]
-                if elapsed < 3:
-                    penalty = float(max(0, len(text) - 40)) / 70
-                    wait = 0.8 + penalty
-                    if elapsed < wait:
-                        time.sleep(wait - elapsed)
+            recipient = self.stack.setdefault(recipient_id, {
```

Comment:
Channeling my inner @dgw with a nitpick: what if, instead of `recipientdict` it was `recipient_dict` or `recipient_stack`? While the 'td' letter combo is usually enough to indicate a _compound word_  (not that recipien**td**ict, is a real word, but... you know what I'm getting at), and pronunciation as two separate words, I believe readability is essential as well, so underscores would help here.

Reply:
updated to `recipient_stack`